tests/files/binary-files.cpp

```diff
@@ -0,0 +1,198 @@
+            0xff, 0xff, 0xff, 0xff, 0xff, 0xfe, 0x1d, 0xc0,
+            0x0,  0x0,  0x0,  0x0,  0x0,  0x1,  0xe2, 0x40,
+            0x42, 0x5, 0x33, 0x33,
+            0xc0, 0x4b, 0xe6, 0x66, 0x66, 0x66, 0x66, 0x66
+        };
+        CHECK(content == expected);
+
+        auto file = BigEndianFile(filename, File::Mode::READ);
+        CHECK(file.read_single_char() == 'A');
+        CHECK(file.read_single_char() == 'B');
+        CHECK(file.read_single_char() == 'C');
+        CHECK(file.read_single_char() == 'D');
+        CHECK(file.read_single_i16() == -42);
+        CHECK(file.read_single_u16() == 42);
+        CHECK(file.read_single_i32() == -573);
+        CHECK(file.read_single_u32() == 573);
+        CHECK(file.read_single_i64() == -123456);
+        CHECK(file.read_single_u64() == 123456);
+        CHECK(file.read_single_f32() == 33.3f);
+        CHECK(file.read_single_f64() == -55.8);
+
+        {
+            auto file = BigEndianFile(filename, File::Mode::APPEND);
+            write_binary_file(file);
+        }
+
+        content = read_binary_file(filename);
+        auto expected_2 = expected;
+        expected_2.insert(expected_2.end(), expected.begin(), expected.end());
+        CHECK(content == expected_2);
+    }
+
+    SECTION("little endian") {
+        auto filename = NamedTempPath(".data");
+        {
+            auto file = LittleEndianFile(filename, File::Mode::WRITE);
+            write_binary_file(file);
+        }
+
+        auto content = read_binary_file(filename);
+        auto expected = std::vector<uint8_t> {
+            'A', 'B', 'C', 'D',
+            0xd6, 0xff,
+            0x2a, 0x0,
+            0xc3, 0xfd, 0xff, 0xff,
+            0x3d, 0x2,  0x0,  0x0,
+            0xc0, 0x1d, 0xfe, 0xff, 0xff, 0xff, 0xff, 0xff,
+            0x40, 0xe2, 0x1,  0x0,  0x0,  0x0,  0x0,  0x0,
+            0x33, 0x33, 0x5,  0x42,
+            0x66, 0x66, 0x66, 0x66, 0x66, 0xe6, 0x4b, 0xc0,
+        };
+        CHECK(content == expected);
+
+        auto file = LittleEndianFile(filename, File::Mode::READ);
+        CHECK(file.read_single_char() == 'A');
+        CHECK(file.read_single_char() == 'B');
+        CHECK(file.read_single_char() == 'C');
+        CHECK(file.read_single_char() == 'D');
+        CHECK(file.read_single_i16() == -42);
+        CHECK(file.read_single_u16() == 42);
+        CHECK(file.read_single_i32() == -573);
+        CHECK(file.read_single_u32() == 573);
+        CHECK(file.read_single_i64() == -123456);
+        CHECK(file.read_single_u64() == 123456);
+        CHECK(file.read_single_f32() == 33.3f);
+        CHECK(file.read_single_f64() == -55.8);
+
+        {
+            auto file = LittleEndianFile(filename, File::Mode::APPEND);
+            write_binary_file(file);
+        }
+
+        content = read_binary_file(filename);
+        auto expected_2 = expected;
+        expected_2.insert(expected_2.end(), expected.begin(), expected.end());
+        CHECK(content == expected_2);
+    }
+}
```

Comment:
AFAIK chemfiles allows writing to newly created files in append mode like it's done [here](https://github.com/chemfiles/chemfiles/blob/6129ac7c06cb6261589818f4fd6b42276a6b9121/tests/formats/trr.cpp#L290).
This does not work for `BinaryFile`:
```cpp
TEST_CASE("Append on new trajectory") {
    SECTION("big endian") {
        auto filename = NamedTempPath(".data");
        {
            auto file = BigEndianFile(filename, File::Mode::APPEND);
            write_binary_file(file);
        }
    }
}
```

Reply:
I've added a bunch of tests for append mode, all the way to the final format implementation (BinaryFile, Netcdf3File, AmberNetCDF), hopefully everything should work now!